searchlib/src/vespa/searchlib/queryeval/blueprint.cpp

```diff
@@ -403,11 +419,13 @@ IntermediateBlueprint::visitMembers(vespalib::ObjectVisitor &visitor) const
 }
 
 void
-IntermediateBlueprint::fetchPostings(bool strict)
+IntermediateBlueprint::fetchPostings(const ExecuteInfo &execInfo)
 {
+    ExecuteInfo childInfo = ExecuteInfo(execInfo.isStrict() && inheritStrict(0), execInfo.hitRate());
     for (size_t i = 0; i < _children.size(); ++i) {
-        bool strictChild = (strict && inheritStrict(i));
-        _children[i]->fetchPostings(strictChild);
+        Blueprint & child = *_children[i];
+        child.fetchPostings(childInfo);
+        childInfo = computeNext(child, ExecuteInfo(execInfo.isStrict() && inheritStrict(i+1), childInfo.hitRate()));
```

Comment:
minor consistency issue: asking if a non-existing child should inherit strictness. (not an actual issue in foreseeable future)